lib/ssl/src/ssl_handshake.erl

```diff
@@ -2648,6 +2724,31 @@ decode_extensions(<<?UINT16(?PRE_SHARED_KEY_EXT), ?UINT16(Len),
                                #pre_shared_key_server_hello{
                                   selected_identity = Identity}});
 
+%% RFC6066, if a server returns a "CertificateStatus" message, then
+%% the server MUST have included an extension of type "status_request"
+%% with empty "extension_data" in the extended server hello.
+decode_extensions(<<?UINT16(?STATUS_REQUEST), ?UINT16(Len),
+                    _ExtensionData:Len/binary, Rest/binary>>, Version,
+                    MessageType = server_hello, Acc)
+  when Len =:= 0 ->
+    decode_extensions(Rest, Version, MessageType,
+                      Acc#{ocsp_stapling => undefined});
```

Comment:
It would be better if the map key is called the same as the decoded extension, in this case `status_request`. This change should be propagated to other parts of the code.
`status_request` itself is overloaded as it can contain either a `CertificateStatusRequest` or `CertificateStatus` (TLS 1.3). There is also a missing encode_extension clause for encoding a `status_request`. It will be used on the server side in the future. With the help of that encode clause we will be able to extend the property tests suite.